tests/Unit/Helpers/Domain/CoordinateMaps/TestMapHelpers.hpp

```diff
@@ -220,6 +282,83 @@ void test_inv_jacobian(
     }
   }
 }
+
+template <typename Map>
+void test_inv_jacobian(const Map& map,
+                       const std::array<DataVector, Map::dim>& test_point) {
+  INFO("Test inverse Jacobian");
+  CAPTURE(test_point);
+  const auto jacobian = map.jacobian(test_point);
+  const auto inv_jacobian = map.inv_jacobian(test_point);
+
+  const auto expected_identity = [&jacobian, &inv_jacobian]() {
+    auto identity =
+        make_with_value<tnsr::Ij<DataVector, Map::dim, Frame::NoFrame>>(
+            get_size(jacobian.get(0, 0)), 0.0);
+    for (size_t i = 0; i < Map::dim; ++i) {
+      for (size_t j = 0; j < Map::dim; ++j) {
+        for (size_t l = 0; l < jacobian.get(0, 0).size(); l++) {
+          for (size_t k = 0; k < Map::dim; ++k) {
+            gsl::at(identity.get(i, j), l) +=
+                gsl::at(jacobian.get(i, k), l) *
+                gsl::at(inv_jacobian.get(k, j), l);
+          }
+        }
+      }
+    }
+    return identity;
+  }();
+
+  for (size_t i = 0; i < Map::dim; ++i) {
+    for (size_t j = 0; j < Map::dim; ++j) {
+      for (size_t k = 0; k < gsl::at(test_point, 0).size(); k++) {
+        CHECK(gsl::at(expected_identity.get(i, j), k) ==
+              approx(i == j ? 1. : 0.));
+      }
+    }
+  }
+}
+
+template <typename Map>
+void test_inv_jacobian(
+    const Map& map, const std::array<DataVector, Map::dim>& test_point,
+    const double time,
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time) {
+  INFO("Test inverse Jacobian");
+  CAPTURE(test_point);
+  CAPTURE(time);
+  const auto jacobian = map.jacobian(test_point, time, functions_of_time);
+  const auto inv_jacobian =
+      map.inv_jacobian(test_point, time, functions_of_time);
+
+  const auto expected_identity = [&jacobian, &inv_jacobian]() {
+    auto identity =
+        make_with_value<tnsr::Ij<DataVector, Map::dim, Frame::NoFrame>>(
+            get_size(jacobian.get(0, 0)), 0.0);
+    for (size_t i = 0; i < Map::dim; ++i) {
+      for (size_t j = 0; j < Map::dim; ++j) {
+        for (size_t l = 0; l < jacobian.get(0, 0).size(); l++) {
+          for (size_t k = 0; k < Map::dim; ++k) {
+            get_element(identity.get(i, j), l) +=
```

Comment:
`gsl::at`

Reply:
I'm getting an error when I try to add this saying `assignment of read only location`.

Reply:
What compiler are you using?  Seems OK with clang 16.